### PR TITLE
Convert all Toast alerts to Modals

### LINKED ISSUE #1037: Convert all Toast alerts to Modals with an OK button to close.
Convert toast messages to modal dialog with an OK button to close.

### PR DESCRIPTION
**Pull request by JustinBeBoy**
Resolve #1037 

This PR custom success, error modals and change most of Toast alerts to modals

**Screenshot**
image

image

### CODE REVIEW

File: ui/page/wallet_settings_page.go [hunk 1]
```diff
@@ -352,15 +355,21 @@ func (pg *WalletSettingsPage) HandleUserInteractions() {
 				PositiveButton(values.String(values.StrRescan), func(isChecked bool) bool {
 					err := pg.WL.MultiWallet.RescanBlocks(pg.wallet.ID)
 					if err != nil {
+						errMess := err.Error()
```
Comment (devchoplife) [inline ui/page/wallet_settings_page.go]:
```suggestion
						errMsg := err.Error()
```

File: ui/page/wallet_settings_page.go [hunk 2]
```diff
@@ -352,15 +355,21 @@ func (pg *WalletSettingsPage) HandleUserInteractions() {
...
 						if err.Error() == dcrlibwallet.ErrNotConnected {
-							pg.Toast.NotifyError(values.String(values.StrNotConnected))
-							return true
+							errMess = values.String(values.StrNotConnected)
```
Comment (devchoplife) [inline ui/page/wallet_settings_page.go]:
```suggestion
							errMsg = values.String(values.StrNotConnected)
```

File: ui/values/localizable/en.go [hunk 3]
```diff
@@ -499,4 +499,8 @@ unlockWithPassword = "Unlock with password"
 "confirmVote" = "Confirm your vote"
 "policySetSuccessfully" = "Your treasury policy has been successfully updated!"
 "colon" = ": "
+"ok" = "OK"
+"dexDataReset" = "DEX client data reset complete."
+"dexDataResetFalse" = "DEX client data reset failed. Check the logs."
+"notSameAccoutMixUnmix" = ""Cannot use same account for mixed & unmixed""
```
Comment (devchoplife) [inline ui/values/localizable/en.go:509]: Please add these to the es.go and fr.go files

**Review by devchoplife — CHANGES_REQUESTED** [review]

File: ui/values/localizable/en.go (hunk 3 above)
Comment (JustinBeBoy) [reply ui/values/localizable/en.go:509]: it doesn't necessarily have to, we would have a translator do all non-english translations

Comment (devchoplife) [reply ui/values/localizable/en.go:509]: Alright fine
Resolve the conversations

**Review by devchoplife — DISMISSED** [review]
LGTM

**Comment by Fegor-Clinton** [comment]
This is a simple function that requires just a toggle switch on or off.
Also, the recent mockups doesn't contain any modal to confirm transaction notification success.
There is a ton of white space in this screenshot, it doesn't look like the initial pixels in which modals are designed with for GoDCR Version 2 was used here.

File: ui/decredmaterial/icon_gallery.go [hunk 4]
```diff
@@ -12,7 +12,7 @@ type Icons struct {
 	ContentAdd, NavigationCheck, NavigationMore, ActionCheckCircle, ActionInfo, NavigationArrowBack,
 	NavigationArrowForward, ActionCheck, ChevronRight, NavigationCancel, NavMoreIcon,
 	ImageBrightness1, ContentClear, DropDownIcon, Cached, ContentRemove, ConcealIcon, RevealIcon,
-	SearchIcon, PlayIcon *widget.Icon
+	SearchIcon, PlayIcon, ErrorIcon *widget.Icon
```
Comment (Sirmorrison) [inline ui/decredmaterial/icon_gallery.go]: this is not needed as the icon used for the modal on the mockup already exist.

image

File: ui/decredmaterial/icon_gallery.go [hunk 5]
```diff
@@ -59,6 +59,7 @@ func (i *Icons) StandardMaterialIcons() *Icons {
 	i.RevealIcon = MustIcon(widget.NewIcon(icons.ActionVisibilityOff))
 	i.SearchIcon = MustIcon(widget.NewIcon(icons.ActionSearch))
 	i.PlayIcon = MustIcon(widget.NewIcon(icons.AVPlayArrow))
+	i.ErrorIcon = MustIcon(widget.NewIcon(icons.AlertError))
```
Comment (Sirmorrison) [inline ui/decredmaterial/icon_gallery.go:59]: not needed.

File: ui/page/seedbackup/verify_seed.go [hunk 6]
```diff
@@ -164,7 +164,10 @@ func (pg *VerifySeedPage) verifySeed() {
 				_, err := pg.WL.MultiWallet.VerifySeedForWallet(pg.wallet.ID, seed, []byte(password))
 				if err != nil {
 					if err.Error() == dcrlibwallet.ErrInvalid {
-						pg.Toast.NotifyError("Failed to verify. Please go through every word and try again.")
+						errModal := modal.NewErrorModal(pg.Load, "Failed to verify. Please go through every word and try again.", func(isChecked bool) bool {
```
Comment (Sirmorrison) [inline ui/page/seedbackup/verify_seed.go:167]: lets localize all string variables used on the modal.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
`bound([2405:4803:fe33:b7a0:a42e:4b85:2fbb:fd51]:19108): dial tcp [2405:4803:fe33:b7a0:a42e:4b85:2fbb:fd51]:19108: connect: no route to host
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x4db30b1]

goroutine 36 [running]:
github.com/planetdecred/godcr/ui/page/components.(*DexServerSelector).startDexClient(0xc000397bf0)
	/Users/sirmorrison/Desktop/go projects/godcr-gio/ui/page/components/dexserver_selector.go:89 +0x1b1
github.com/planetdecred/godcr/ui/page/components.(*DexServerSelector).Expose(...)
	/Users/sirmorrison/Desktop/go projects/godcr-gio/ui/page/components/dexserver_selector.go:49
github.com/planetdecred/godcr/ui/page.(*WalletDexServerSelector).OnNavigatedTo(0xc0000acd80)
	/Users/sirmorrison/Desktop/go projects/godcr-gio/ui/page/wallet_dex_selector_page.go:73 +0x7b
github.com/planetdecred/godcr/app.(*PageStack).Reset(0xc0000eaa50, {0xc0001dc2a0, 0x1, 0x1})
	/Users/sirmorrison/Desktop/go projects/godcr-gio/app/pagestack.go:153 +0x112
github.com/planetdecred/godcr/app.(*SimpleWindowNavigator).ClearStackAndDisplay(0xc0000eaa80, {0x58ed270, 0xc0000acd80})
	/Users/sirmorrison/Desktop/go projects/godcr-gio/app/window.go:84 +0x8f
github.com/planetdecred/godcr/ui/page.(*startPage).openWallets(0xc0020d7c00, {0x0, 0xc000472d00})
	/Users/sirmorrison/Desktop/go projects/godcr-gio/ui/page/start_page.go:104 +0x16e
created by github.com/planetdecred/godcr/ui/page.(*startPage).OnNavigatedTo
	/Users/sirmorrison/Desktop/go projects/godcr-gio/ui/page/start_page.go:59 +0xd1`

the app crash this log.. 

The Pr also need rebase.

And the Modal size used for the notification seems larger than the mockup as well.

**Pull request closed without merging**